### PR TITLE
Add retry to batcher

### PR DESCRIPTION
**Pull request by hkalodner**

### CODE REVIEW

**Review by joshuacolvin0 — APPROVED** [review]
LGTM

**Review by joshuacolvin0 — APPROVED** [review]
LGTM

File: packages/arb-util/broadcastclient/broadcastclient.go [hunk 1]
```diff
@@ -164,7 +153,11 @@ func (bc *BroadcastClient) RetryConnect(ctx context.Context, messageReceiver cha
 		}
 
 		bc.RetryCount++
-		_ = bc.connect(ctx, messageReceiver)
+		_, err := bc.connect(ctx, messageReceiver)
+		if err == nil {
+			bc.retrying = false
```
Comment (PlasmaPower) [inline packages/arb-util/broadcastclient/broadcastclient.go:158]: This retrying field seems to never be used as far as I can tell

**Review by PlasmaPower — CHANGES_REQUESTED** [review]
Other than that, LGTM

File: packages/arb-util/broadcastclient/broadcastclient.go (hunk 1 above)
Comment (hkalodner) [reply packages/arb-util/broadcastclient/broadcastclient.go:158]: Yeah it's never used except I think in a test. This whole component needs some work, so this PR just needs to get it in a stable state

**Pull request merged**